src/darker/tests/test_git.py

```diff
@@ -403,6 +403,30 @@ def test_git_get_content_at_revision_stderr(git_repo, capfd, caplog):
     assert caplog.text == ""
 
 
+def test_git_get_content_at_revision_encoding(git_repo):
+    """Git file is loaded using its historical encoding"""
+    code = "foo = 'é'"
+    commits: List[Tuple[str, str, Tuple[str, ...]]] = []
+
+    git_repo.add({"file.py": code + "\n"}, commit="Default encoding")
+    commit = git_repo.get_hash()[:7]
+    encoding = "utf-8"
+    lines: Tuple[str, ...] = (code,)
+    commits.append((commit, encoding, lines))
+    for encoding in ["iso-8859-1", "utf-8"]:
+        lines = (f"# coding: {encoding}", code)
+        content = ("\n".join(lines) + "\n").encode(encoding)
+        git_repo.add({"file.py": content}, commit=f"{encoding}")
+        commit = git_repo.get_hash()[:7]
+        commits.append((commit, encoding, lines))
+    commits.append((":WORKTREE:", encoding, lines))
+
+    for commit, encoding, lines in commits:
+        result = git.git_get_content_at_revision(Path("file.py"), commit, git_repo.root)
+        assert result.lines == lines
+        assert result.encoding == encoding
```

Comment:
If the test for one of the commits fails, the remaining commits won't be tested at all. That's a bit contrary to unit test best practices.

Maybe populate the repository in a fixture, and turn the `for` loop assertions into a parameterized test?

Or, we could take [pytest-assume](https://github.com/astraw38/pytest-assume) into use.